exchanges/btse/btse_websocket.go

```diff
@@ -70,119 +68,91 @@ func (b *BTSE) WsHandleData() {
 				continue
 			}
 
-			msgType := MsgType{}
-			err = common.JSONDecode(resp.Raw, &msgType)
+			result := Result{}
+			err = common.JSONDecode(resp.Raw, &result)
 			if err != nil {
+				log.Println("\n\n\n\nWTF\n")
 				b.Websocket.DataHandler <- err
 				continue
 			}
-			switch msgType.Type {
-			case "ticker":
-				var t wsTicker
-				err = common.JSONDecode(resp.Raw, &t)
+			switch {
+			case strings.Contains(result["topic"].(string), "tradeHistory"):
+				var tradeHistory wsTradeHistory
+				err := common.JSONDecode(resp.Raw, &tradeHistory)
 				if err != nil {
 					b.Websocket.DataHandler <- err
 					continue
 				}
-				p := strings.Replace(t.Price.(string), ",", "", -1)
-				price, err := strconv.ParseFloat(p, 64)
+				b.Websocket.DataHandler <- tradeHistory
+			case strings.Contains(result["topic"].(string), "orderBookApi"):
+				var t wsOrderBook
+				err = common.JSONDecode(resp.Raw, &t)
 				if err != nil {
 					b.Websocket.DataHandler <- err
 					continue
 				}
-
-				b.Websocket.DataHandler <- wshandler.TickerData{
-					Timestamp: time.Now(),
-					Pair:      currency.NewPairDelimiter(t.ProductID, "-"),
-					AssetType: orderbook.Spot,
-					Exchange:  b.GetName(),
-					OpenPrice: price,
+				var asks, bids []orderbook.Item
+				for i := range t.Data.BuyQuote {
+					var tempAsks orderbook.Item
+					p := strings.Replace(t.Data.BuyQuote[i].Price, ",", "", -1)
+					price, err := strconv.ParseFloat(p, 64)
+					if err != nil {
+						b.Websocket.DataHandler <- err
+					}
+					a := strings.Replace(t.Data.BuyQuote[i].Size, ",", "", -1)
+					amount, err := strconv.ParseFloat(a, 64)
+					if err != nil {
+						b.Websocket.DataHandler <- err
+					}
+					tempAsks.Amount = amount
+					tempAsks.Price = price
+					asks = append(asks, tempAsks)
 				}
-			case "snapshot":
-				snapshot := websocketOrderbookSnapshot{}
-				err := common.JSONDecode(resp.Raw, &snapshot)
-				if err != nil {
-					b.Websocket.DataHandler <- err
-					continue
+				for j := range t.Data.SellQuote {
+					var tempBids orderbook.Item
+					p := strings.Replace(t.Data.SellQuote[j].Price, ",", "", -1)
+					price, err := strconv.ParseFloat(p, 64)
+					if err != nil {
+						b.Websocket.DataHandler <- err
+					}
+					a := strings.Replace(t.Data.SellQuote[j].Size, ",", "", -1)
+					amount, err := strconv.ParseFloat(a, 64)
+					if err != nil {
+						b.Websocket.DataHandler <- err
+					}
+					tempBids.Amount = amount
+					tempBids.Price = price
+					bids = append(bids, tempBids)
 				}
-
-				err = b.wsProcessSnapshot(&snapshot)
+				var newOB orderbook.Base
+				newOB.Asks = asks
+				newOB.Bids = bids
+				newOB.AssetType = orderbook.Spot
+				pair := strings.Replace(t.Topic, "orderBookApi:", "", 1)
```

Comment:
Can be simplified to `pair := currency.NewPairFromString(t.Topic[strings.Index(t.Topic, ":")+1:strings.Index(t.Topic, "_")])`